src/components/new-user-form/index.tsx

```diff
@@ -0,0 +1,211 @@
+import { ApolloError } from "@apollo/client";
+import React, { FormEvent, useState } from "react";
+import { formatPhone } from "../../utils/phone-formatter";
+import { validateBirthDate } from "../../validators/birthdate";
+import { validateConfirmPassword } from "../../validators/confirm-password";
+import { validateEmail } from "../../validators/email";
+import { validateName } from "../../validators/name";
+import { validatePassword } from "../../validators/password";
+import { validatePhone } from "../../validators/phone";
+import { Button } from "../button";
+import { Input } from "../input";
+import * as S from "./style";
+
+interface NewUserParamsTypes {
+  variables: {
+    data: {
+      name: string;
+      email: string;
+      phone: string;
+      birthDate: string;
+      password: string;
+      role: string;
+    };
+  };
+}
```

Comment:
duas coisas aqui:
1) como o form só tem que se preocupar com o que diz respeito ao form ([S do SOLID](https://konstantinlebedev.com/solid-in-react/)), esse 'retorno' não precisa estar dentro do `variables` nem do `data`. isso deixa para quem for fazer a chamada do graphql 👍 
```tsx
<NewUserForm 
  onSubmitSuccess={(formData) => saveUserMutation({ variables: { data: formData } })}
/>
```
2) renomear para `NewUserFormData`? 
```suggestion
// 'export' pq quem for usar o Form provavelmente vai usar essa interface tb
export interface NewUserFormData { 
      name: string;
      email: string;
      phone: string;
      birthDate: string;
      password: string;
      role: string;
}
```

Reply:
1. Perfeito, pensando dessa forma faz muito mais sentido deixar o componente lidar apenas com o que diz respeito ao form e delegar a tarefa da mutation à quem faz a chamada ao invés de já tratar dados no próprio componente. Corrigido!

2. Corrigido, segui sua sugestão

Obrigado Beto!!